aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -311,6 +311,68 @@ void bernoulli_mkl_kernel(Tensor &self, const double p, Generator* gen) {
 }
 #endif
 
+#if !AT_MKL_ENABLED()
+void uniform_mkl_kernel(Tensor & self, const double from, const double to, Generator* gen) {
+  // Use AT_ASSERTM because this should never be reached, and AT_ASSERTM tells
+  // users to report this as a bug.
+  AT_ASSERTM(false, "ATen not compiled with MKL");
+}
+#else
+
+static inline void vslRngUniform(const int method, VSLStreamStatePtr stream,
+    const int n, float* r, const float a, const float b) {
+  vsRngUniform( method, stream, n, r, a, b );
+}
+
+static inline void vslRngUniform(const int method, VSLStreamStatePtr stream,
+    const int n, double* r, const double a, const double b) {
+  vdRngUniform( method, stream, n, r, a, b );
+}
+
+void uniform_mkl_kernel(Tensor & self, const double from, const double to, Generator* gen) {
```

Comment:
Do you need two copies of this code compiled (AVX on/off)?

Reply:
Not really, I just aligned `uniform_mkl_kernel` with `bernoulli_mkl_kernel` in the same manner.